### PR TITLE
Feature/2268 - create a data source plugin for mailgun

### PR DESCRIPTION
**Pull request by mslourens**
Created an API plugin for Mailgun. I don't have a Mailgun domain or payed account, so I could not fully test it myself. Maybe @Navaneeth-pk can test it. The implementation is based on the SendGrid plugin, which is quite similar. 

Implements #2268 

### Configuration
image

### CODE REVIEW

**Comment by Navaneeth-pk** [comment]
Thanks @mslourens! 
We will test this PR tomorrow.

File: docs/docs/data-sources/mailgun.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Optional parameters:
+- Body as HTML
+
+<img class="screenshot-full" src="/img/datasource-reference/MailGun/MailGun-query.jpg" alt="ToolJet - Query MailGun" height="420"/>
+
+:::info
+**Send mail to** - accepts an array/list of emails separated by comma.
+For example:
+`{{["dev@tooljet.io", "admin@tooljet.io"]}}`.
```
Comment (gsmithun4) [inline docs/docs/data-sources/mailgun.md:46]: It supports single mail id
`dev@tooljet.io`
please mention here

**Comment by gsmithun4** [comment]
Deployed https://tj-mailgun.herokuapp.com/
and tested, working fine

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
```
[the remark this answers is not in the document]
Comment (mslourens) [reply .vscode/launch.json]: I removed them and added .`vsocde/*` to gitignore, to prevent adding IDE related files to the repo.

[the remark this answers is not in the document]
Comment (Navaneeth-pk) [reply .vscode/launch.json]: The existing files in `.vscode/` directory are needed, they are being used for linting, etc. So adding to gitignore is not a good idea. If you are proposing changes to the vscode config, please open another issue as it is not related to the Mailgun plugin. Thanks!
@mslourens

File: plugins/packages/mailgun/lib/index.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    if (queryOptions.html && queryOptions.html.length > 0) {
+      emailOptions.html = queryOptions.html;
+    }
+
+    try {
+      result = await mailGunClient.messages.create(sourceOptions.domain, emailOptions);
+    } catch (error) {
+      console.error(error.response);
```
Comment (gsmithun4) [inline plugins/packages/mailgun/lib/index.ts:34]: Please print the whole `error` object
`console.error(error);`

File: plugins/packages/smtp/package.json [hunk 4]
```diff
@@ -20,8 +20,5 @@
     "@tooljet-plugins/common": "file:../common",
     "nodemailer": "^6.7.2",
     "react": "^17.0.2"
-  },
```
Comment (gsmithun4) [inline plugins/packages/smtp/package.json:23]: Can you please revert this change and create new PR for this?

File: frontend/package-lock.json [hunk 5]
```diff
@@ -133,6 +134,7 @@
         "@tooljet-plugins/typesense": "file:packages/typesense"
       },
       "devDependencies": {
+        "@types/nodemailer": "^6.4.4",
```
Comment (gsmithun4) [inline frontend/package-lock.json:137]: Do we need nodemailer as devDependency in front end?

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 6]
```diff
@@ -34,3 +34,7 @@
 /docs/.firebaserc
 /node_modules
 /frontend/cypress/videos
+
+# Ignore IDE config files
+.vscode/*
```
Comment (Navaneeth-pk) [inline .gitignore:39]: Please remove this entry. The files under this directory are being used by developers who use vscode, mostly for linting.

File: frontend/package-lock.json (hunk 5 above)
Comment (mslourens) [reply frontend/package-lock.json:137]: No, it was added by accident

File: .gitignore (hunk 6 above)
Comment (mslourens) [reply .gitignore]: Normally I add IDE specific config files to my gitignore, because every developer uses his own configs/settings. I removed this line.

Comment (Navaneeth-pk) [reply .gitignore]: You are right. We can discuss it as a separate issue to remove .vscode directory from gitignore since it is not related to this PR.

File: frontend/package-lock.json (hunk 5 above)
Comment (gsmithun4) [reply frontend/package-lock.json:137]: Can you please fix it?

File: frontend/package-lock.json [hunk 7]
```diff
@@ -22431,6 +22433,7 @@
         "@tooljet-plugins/stripe": "file:packages/stripe",
         "@tooljet-plugins/twilio": "file:packages/twilio",
         "@tooljet-plugins/typesense": "file:packages/typesense",
+        "@types/nodemailer": "^6.4.4",
```
Comment (gsmithun4) [inline frontend/package-lock.json:22436]: @mslourens One more here

**Comment by Navaneeth-pk** [comment]
`.vscode/extension.json` & `.vscode/settings.json` is removed in this PR, please revert those changes.
@mslourens

**Review by gsmithun4 — APPROVED** [review]
LGTM

**Comment by mslourens** [comment]
> `.vscode/extension.json` & `.vscode/settings.json` is removed in this PR, please revert those changes. @mslourens

I reverted the changes. Sorry for that, I won't touch those files again ;)

**Comment by Navaneeth-pk** [comment]
> I won't touch those files again ;)
Thanks for your patience :) Just trying to make sure every PR has only related changes. We can open another issue for cleaning up unwanted files if needed.

**Comment by Navaneeth-pk** [comment]
There are some lint errors since we've merged https://github.com/ToolJet/ToolJet/pull/2402. Merging this PR and will fix the lint errors in a separate PR as there are errors in files that are not related.

**Review by Navaneeth-pk — APPROVED** [review]
